libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -102,9 +102,16 @@ bool HTTPClient::begin(WiFiClient &client, const String& url) {
  */
 bool HTTPClient::begin(WiFiClient &client, const String& host, uint16_t port, const String& uri, bool https)
 {
+    // Disconnect when reusing HTTPClient to talk to a different host
+    if (_host != "" && _host != host) {
```

Comment:
Here and below, can you please change to
`_host != emptyString` or better: `_host.length() > 0`

Reply:
or `!_host.isEmpty()`